File: frameworks/domain.core/src/main/java/com/example/frameworks/domain/core/Page.java
package com.example.frameworks.domain.core;

import java.util.List;
import java.util.function.Function;

public interface Page<T> {
    List<T> getContent();
    long getTotalElements();
    int getSize();
    int getNumber();
    int getNumberOfElements();
    <U> Page<U> map(Function<? super T, ? extends U> converter);
}

File: frameworks/domain.core/src/main/kotlin/com/example/frameworks/domain/core/Page.kt
package com.example.frameworks.domain.core

import java.util.function.Function

interface Page<T> {
    fun content(): List<T>
    fun size(): Int
    fun number(): Int
    fun numberOfElements(): Int
    fun numberOfTotalPages(): Int
    fun numberOfTotalElements(): Long
    fun first(): Boolean
    fun last(): Boolean
    fun sort(): Sort
    fun <U> map(converter: Function<in T, out U>): Page<U>
}
